From 779d03945de400b2e0edf864a95ee75cb8d7ced1 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Mon, 27 Jan 2025 00:54:28 +0800
Subject: [PATCH 3/5] AOSCOS: Settings: drop all unusable hardware decoding
 options

Upstream commit 5e4507df42da ("PlaybackSettings: remove some hardware
decoding options") removed a lot of hardware decoding options that were
deemed unnecessarily. I argue that even more should be removed, namely
d3d11va (Windows-specific) and videotoolbox (macOS-specific).

This seems like a platform-agnostic and hard-coded list anyway, so change
it to suit our reality.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 src/qml/Settings/PlaybackSettings.qml | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/src/qml/Settings/PlaybackSettings.qml b/src/qml/Settings/PlaybackSettings.qml
index c213bf9d..0af45401 100644
--- a/src/qml/Settings/PlaybackSettings.qml
+++ b/src/qml/Settings/PlaybackSettings.qml
@@ -191,27 +191,10 @@ SettingsBasePage {
                 id: hwDecModel
                 ListElement { key: "auto"; }
                 ListElement { key: "auto-safe"; }
-                ListElement { key: "auto-copy"; }
                 ListElement { key: "vdpau"; }
-                ListElement { key: "vdpau-copy"; }
                 ListElement { key: "vaapi"; }
-                ListElement { key: "vaapi-copy"; }
-                ListElement { key: "videotoolbox"; }
-                ListElement { key: "videotoolbox-copy"; }
-                ListElement { key: "dxva2"; }
-                ListElement { key: "dxva2-copy"; }
-                ListElement { key: "d3d11va"; }
-                ListElement { key: "d3d11va-copy"; }
-                ListElement { key: "mediacodec"; }
-                ListElement { key: "mediacodec-copy"; }
-                ListElement { key: "mmal"; }
-                ListElement { key: "mmal-copy"; }
                 ListElement { key: "nvdec"; }
-                ListElement { key: "nvdec-copy"; }
                 ListElement { key: "cuda"; }
-                ListElement { key: "cuda-copy"; }
-                ListElement { key: "crystalhd"; }
-                ListElement { key: "rkmpp"; }
             }
 
             onActivated: {
-- 
2.48.1

